### PR TITLE
Benchmarking workflows/scripts

### PR DESCRIPTION
**Pull request by paradust7**
- track_minetest workflow: Periodically updates the minetest/irrlicht revision numbers. (pushes back into this repo)
- perf: Builds and runs the benchmarks on every push.

### CODE REVIEW

File: .github/workflows/perf.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      # Checkout irrlicht
+      - uses: actions/checkout@v2
+        with:
+          repository: '${{ github.repository_owner }}/irrlicht'
+          ref: ${{ env.IRRLICHT_REV }}
+          path: 'minetest/lib/irrlichtmt'
+      - name: Install deps
+        run: |
+          pushd minetest
+          source ./util/ci/common.sh --no-irr
```
Comment (sfan5) [inline .github/workflows/perf.yml:43]: why not save yourself the work and use the irrlicht installed by the CI script?

File: .github/workflows/track_minetest.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: track_minetest
+
+# build on c/cpp changes or workflow changes
+on:
+  schedule:
+    # This cron line says to run every minute, but in practice github appears
+    # to run it sporadically every 7 to 15 minutes.
+    - cron: '* * * * *'
```
Comment (sfan5) [inline .github/workflows/track_minetest.yml]: too often is wasteful IMO, how about every 3 hours?

File: .github/workflows/perf.yml (hunk 1 above)
Comment (paradust7) [reply .github/workflows/perf.yml:43]: Thinking ahead to maybe performance testing the client. That would involve rendering frames in a headless environment.

Comment (paradust7) [reply .github/workflows/perf.yml:43]: Also in order to performance test say, irrArray, in a meaningful way, would require up to date irrlicht

File: .github/workflows/track_minetest.yml (hunk 2 above)
Comment (paradust7) [reply .github/workflows/track_minetest.yml]: Does this use up some kind of resource? The check itself is very fast, only takes about 10 seconds. What takes a while is for github to allocate a VM to run it.

Comment (sfan5) [reply .github/workflows/track_minetest.yml]: There's a limit on concurrent jobs per organization: https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration
I don't expect this to be a problem currently but I think it's better to be preemptive.

File: .github/workflows/perf.yml (hunk 1 above)
Comment (sfan5) [reply .github/workflows/perf.yml:43]: Headless rendering will work with the existing setup, but testing the latest commits is a fair point.

File: .github/workflows/track_minetest.yml (hunk 2 above)
Comment (paradust7) [reply .github/workflows/track_minetest.yml]: changed to every 3 hours

**Review by sfan5 — APPROVED** [review]
Let's see how this works

**Comment by sfan5** [comment]
For some reason the automated commits do not actually trigger workflow runs?!

**Comment by paradust7** [comment]
I thought I had tested this, but apparently not:

https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow

To prevent infinite recursion, workflows can't be triggered by commits using a GITHUB_TOKEN belonging to another workflow.
They suggest using a personal API token. I was hoping to avoid this, because it appears there's no way to limit access to a single repo. An API token for a given user would be able to commit to all repos the user has access to. I suppose it would be possible to create a dummy user that only has commit access to that one repo, and then generate a personal API token from that. But that's a lot of extra work that can't be simply written into the workflow yml.

Let me try to combine the two workflows, so that the cron job itself will do the build.

**Comment by sfan5** [comment]
This should work though? https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow
(unless that requires a personal token for the same reasons)

**Comment by paradust7** [comment]
Unfortunately it appears that API endpoint is not accessible either except through a personal token.

As an quick fix, I've just combined the two workflows, so that the build, benchmark, and publish will happen every 3 hours, or every manual push: https://github.com/minetest/benchmarks/pull/2

**Pull request merged**